### PR TITLE
Text and Speech Prenets

### PR DESCRIPTION
**Pull request by liaojh1998**
- Added text and speech prenets in `module.py`.
- The notebook `testing_prepostnet.ipynb` shows an example of them.

Please see `testing_prepostnet.ipynb` for an example using the LJSpeech dataset. I'll add the Postnets later.

This is not built on top of Lucas's branch, but it may have merge conflicts with what his code as this changed `module.py`.

### CODE REVIEW

File: src/module.py [hunk 1]
```diff
@@ -3,22 +3,157 @@
...
+        self.linear_layer = nn.Linear(in_dim, out_dim, bias=bias)
+
+        nn.init.xavier_uniform_(
+            self.linear_layer.weight,
+            gain=nn.init.calculate_gain(w_init))
+
+    def forward(self, x):
+        return self.linear_layer(x)
+
+class Conv(nn.Module):
```
Comment (Lucaskabela) [inline src/module.py:37]: I am not a big fan of these modules, as they essentially just wrap the pytorch modules and use xaiver_uniform instead of Kaiming

Can we rewrite these pre/post net to just directly use the `nn.*` modules instead?  Or perhaps make that a TODO for later - regardless I think we need to consolidate on an approach and I prefer just using what pytorch has to offer over this wrapping approach

**Review by Lucaskabela — CHANGES_REQUESTED** [review]

File: src/module.py (hunk 1 above)
Comment (Lucaskabela) [reply src/module.py:37]: This isn't really a blocker, so in the interest of making progress on the general training loop I am going to proceed with merging this

**Review by Lucaskabela — APPROVED** [review]

File: src/module.py (hunk 1 above)
Comment (liaojh1998) [reply src/module.py:37]: I added these because it seems that the Postnets also use them. Nonetheless, I don't think it would make a big difference if we used Kaiming initialization instead of Xavier. @billyang98 what do you think?

File: src/module.py [hunk 2]
```diff
@@ -3,22 +3,157 @@
...
 
 class SpeechPrenet(nn.Module):
-    # TODO: Fill in from TTS repo :) 
-    def __init__(self):
+    """
+    Prenet for Speech Encoder copied from Transformer-TTS.
+    
+    As described in Ren's paper, 2-layer dense-connected network
+    with hidden size of 256, and the output dimension equals to
+    the hidden size of Transformer. Note that the dropout is
```
Comment (billyang98) [inline src/module.py:76]: can we add dropout?

Comment (liaojh1998) [reply src/module.py:76]: Yeah, this will be added back in. It's in the my other branch, I'll make a PR now,
